src/interfaces.ts

```diff
@@ -20,77 +17,60 @@ export interface ConnectionOptions {
   signer?: Signer;
   host?: string;
   network?: string;
+  contract?: string;
 }
 
 export interface RpcParams {
-  controller?: boolean;
-  createStatement?: string;
-  description?: string;
-  dryrun?: boolean;
-  statement?: string;
-  tableId?: string;
-}
-
-export interface RpcRequestParam {
   controller?: string;
   /* eslint-disable-next-line camelcase */
   create_statement?: string;
-  description?: string;
-  dryrun?: boolean;
-  id?: string;
   statement?: string;
-}
-
-export interface ConnectionReceipt {
-  jwsToken: Token;
-  ethAccounts: Array<string>;
+  /* eslint-disable-next-line camelcase */
+  txn_hash?: string;
 }
 
 export interface SupportedNetwork {
   key: string;
+  name: string; // Matches naming convention used in https://chainlist.org and ethers network lib
   phrase: string;
-}
-
-/**
- * ColumnDescriptor gives metadata about a colum (name, type)
- */
-export interface ColumnDescriptor {
-  name: string;
+  chainId: number;
 }
 
 export type KeyVal<T = any> = [string, T];
 
-export type Column = Array<ColumnDescriptor>;
-
-export type Row = Array<string | number | boolean>;
+export type Columns = Array<{ name: string }>;
 
-export interface ReadQueryResult {
-  columns: Array<Column>;
-  rows: Array<Row>;
+export interface ReadQueryResult<
+  Rows extends Array<string | number | boolean> = Array<any>
+> {
+  columns: Columns;
+  rows: Rows;
 }
 
-export interface CreateTableOptions {
-  /** A human readable description of the nature and purpoe of the table */
-  description?: string;
-  /** If your table was minted, but never created on tableland, use this param to create it. */
-  id?: string;
-  /** do a dry run of create to see if the create statement is valid without creating the table */
-  dryrun?: boolean;
+export interface WriteQueryResult {
+  hash: string;
 }
 
-export interface CreateTableReceipt {
-  name: string;
-  structureHash: string;
-  description?: string;
+export interface ReceiptResult {
+  /* eslint-disable-next-line camelcase */
+  chain_id: number;
+  /* eslint-disable-next-line camelcase */
+  txn_hash: string;
+  /* eslint-disable-next-line camelcase */
+  block_number: number;
+  error?: string;
 }
 
 export interface StructureHashReceipt {
   structureHash: string;
 }
 
-export interface TableRegistrationReceipt {
-  receipt: ContractReceipt;
-  tableId: BigNumber;
+export interface CreateTableReceipt {
```

Comment:
Check me on this. I've created this as an output to create. Check out the new function sig and tests.